DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,210 @@
+# Declaring Dependencies on Google Java Libraries
+
+Google maintains a number of open source Java libraries that make it
+easier to use services in Google Cloud Platform (GCP). Additionally
+Google maintains several foundational libraries that can be used for
+very general purposes, which the GCP libraries also depend on. This
+document explains how to use bills of materials (BOMs) and other
+strategies to avoid dependency conflicts in these libraries.
```

Comment:
perhaps "avoid dependency conflicts in these libraries" --> "reduce dependency conflicts."

Reply:
I feel like "avoid" is better than "reduce" here - it is a stronger statement, and represents what we're aiming for.

If we remove "in these libraries", then this sentence becomes detached from the prior one - this part links them together.